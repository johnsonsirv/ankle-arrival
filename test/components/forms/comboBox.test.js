/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import ComboBox from '../../../src/components/forms/inputs/comboBox';

function setup() {
  const props = {
    name: 'combo',
    onChange: jest.fn(),
  };
  const options = [
    {
      id: 1,
      value: 'Sample Option 1',
    },
    {
      id: 2,
      value: 'Sample Option 2',
    },
  ];
  const enzymeWrapper = mount(
    <ComboBox {...props}>
      {options.map(item => (
        <option key={item.id} value={item.value}>
          {item.value}
        </option>
      ))}
    </ComboBox>
  );

  return { props, enzymeWrapper };
}

describe('<ComboBox /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 editable <input> select', () => {
    expect(enzymeWrapper.find('select')).toHaveLength(1);
  });
  it('should render 2 <option> elements', () => {
    expect(enzymeWrapper.find('option')).toHaveLength(2);
  });
});

describe('<ComboBox /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onChange', () => {
    enzymeWrapper.simulate('change', {
      target: { value: 'Sample Option 2' },
    });
    expect(props.onChange).toHaveBeenCalled();
  });
});
