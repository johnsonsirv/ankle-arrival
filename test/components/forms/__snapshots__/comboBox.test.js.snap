// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComboBox /> rendering renders correctly 1`] = `ReactWrapper {}`;
